Modules/System/Location/src/Location.Page.al

```diff
@@ -0,0 +1,160 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary>
+/// Provides an interface for accessing the location on the client device.
+/// </summary>
+/// <example>
+/// <code>
+/// Location.RunModal();
+/// if Location.HasLocation() then begin
+///     Location.GetLocation(Latitude, Longitude);
+/// ...
+/// end;
+/// Clear(Location);
+/// </code>
+/// </example>
+page 50105 "Location"
+{
+    Caption = 'Location request';
+    PageType = Card;
+    Editable = false;
+    LinksAllowed = false;
+    Extensible = true;
+
+    layout
+    {
+        area(content)
+        {
+            group(Location)
+            {
+                Caption = 'Requesting location...';
+                InstructionalText = 'Please, confirm that Business Central can access the location of the device.';
+                Visible = LocationAvailable;
+            }
+            group(LocationNotSupported)
+            {
+                Caption = 'Could not access the location';
+                InstructionalText = 'Could not access the location of the device. Make sure that you are using the app for Windows, Android, or iOS.';
+                Visible = NOT LocationAvailable;
+            }
+        }
+    }
+
+    var
+        LocationPageImpl: Codeunit "Location Page Impl.";
+        [RunOnClient]
+        [WithEvents]
+        LocationProvider: DotNet LocationProvider;
+        LocationAvailable: Boolean;
+
+    /// <summary>
+    /// When the page opens it request the location from the client and shows a view.
+    /// After getting the location, the page will close automatically.
+    /// </summary>
+    trigger OnOpenPage()
+    begin
+        LocationPageImpl.LocationInteractionOnOpenPage(LocationProvider, LocationAvailable);
+    end;
+
+    /// <summary>
+    /// Checks if the location is available on the client device.
+    /// </summary>
+    /// <returns>True if the location is available; false otherwise.</returns>
+    procedure IsAvailable(): Boolean
```

Comment:
You can try and completely "hide" the Location page.

Reply:
> Avoid functions on pages and these should be on the facade.

I am using the same pattern as the camera module. Moving the functions away from the page to the facade is non-trivial in this case, since the page needs to hold the DotNet LocationProvider variable and also receives the location from the async call. You can't move the LocationOptions separately since they are use by the LocationProvider.

> You can try and completely "hide" the Location page.

I am not sure how I would do that. The DotNet LocationProvider needs to be on a page to request the location from the user - and I can't find an option to hide that page.